### PR TITLE
2760 document suc upgrade recovery

### PR DESCRIPTION
**Pull request by jimmykarily**
Part of https://github.com/kairos-io/kairos/issues/2760

### CODE REVIEW

File: content/en/docs/Upgrade/system-upgrade-controller.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+metadata:
+  name: git
+spec:
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+        - name: git
+          image: alpine/git
+          command: ["git", "clone", "https://github.com/rancher/system-upgrade-controller", "/homedir/system-upgrade-controller"]
```
Comment (jimmykarily) [inline content/en/docs/Upgrade/system-upgrade-controller.md:33]: Maybe we want to point to a specific tag/release here?

**Comment by jimmykarily** [comment]
When I upgrade the recovery image, it falls into a loop:

```
kube-system      traefik-8dc7cf49b-kpq2j                                           0/1     Pending   0             3m51s
system-upgrade   apply-os-upgrade-on-localhost-with-8a5cfbc577a39382877df5-hnkt6   1/1     Running   0             16s
system-upgrade   apply-os-upgrade-on-localhost-with-8a5cfbc577a39382877df5-nbmc5   0/1     Unknown   0             3m57s
system-upgrade   apply-os-upgrade-on-localhost-with-8a5cfbc577a39382877df5-zqxr5   0/1     Unknown   0             84s
system-upgrade   system-upgrade-controller-567584c4cc-m9mlt                        1/1     Running   2 (26s ago)   20m
```
everytime the node boots, the upgrade pod starts again and it upgrades recovery, then reboots and all over again.

I'm suspecting that when the node is rebooted, the Plan doesn't have enough time to become "complete" and thus the controller restarts it as soon as the node boots.

**Comment by jimmykarily** [comment]
We facing this issue: https://github.com/rancher/system-upgrade-controller/issues/23#issuecomment-583003758

```
This is exactly the issue: when the process is re-run and has nothing to do it should exit successfully. This is the underlying assumption that this controller was built with: re-application of an upgrade will result in the same end-state (idempotency) AND such will exit successfully.
```

**Comment by jimmykarily** [comment]
Needs the framework to be bumped in kairos for this to work

File: content/en/docs/Upgrade/system-upgrade-controller.md (hunk 1 above)
Comment (jimmykarily) [reply content/en/docs/Upgrade/system-upgrade-controller.md]: templetized it

**Review by Itxaka — APPROVED** [review]

**Pull request merged**